pkg/reconciler/taskrun/validation.go

```diff
@@ -85,3 +105,217 @@ func extractPlatform(tr *tektonapi.TaskRun) (string, error) {
 	}
 	return "", errMissingPlatformParameter
 }
+
+// validateNumericValue validates a string represents a valid integer within the specified range
+// Validation rules:
+// - Must be a valid integer parseable by strconv.Atoi
+// - Must be between 1 and maxValue inclusive
+//
+// Parameters:
+// - value: The string value to validate
+// - maxValue: The maximum allowed value
+//
+// Returns:
+// - int: The validated integer value
+// - error: errInvalidNumericValue if value is invalid or out of range
+func validateNumericValue(value string, maxValue int) (int, error) {
+	num, err := strconv.Atoi(value)
+	if err != nil || num != max(0, min(num, maxValue)) {
+		return 0, errInvalidNumericValue
+	}
+	return num, nil
+}
+
+// validateMaxInstances validates a string represents a valid max-instances value
+// Uses the global maxInstancesValue as the upper limit
+//
+// Parameters:
+// - value: The string value to validate
+//
+// Returns:
+// - int: The validated integer value
+// - error: errInvalidNumericValue if value is invalid or out of range
+func validateMaxInstances(value string) (int, error) {
+	return validateNumericValue(value, maxInstancesValue)
+}
+
+// validateMaxAllocationTimeout validates a string represents a valid allocation timeout value
+// Uses the global maxAllocationTimeout as the upper limit
+//
+// Parameters:
+// - value: The string value to validate
+//
+// Returns:
+// - int: The validated integer value
+// - error: errInvalidNumericValue if value is invalid or out of range
+func validateMaxAllocationTimeout(value string) (int, error) {
+	return validateNumericValue(value, maxAllocationTimeout)
+}
+
+// obfuscateIP takes an IP address and returns it with the first three octets replaced with asterisks
+// Example: "192.168.1.1" becomes "***.***.***.1"
+func obfuscateIP(ip string) string {
+	parts := strings.Split(ip, ".")
+	return "***.***.***." + parts[3]
+}
+
+// validateIP validates that a string represents a valid and reachable IP address
+// Uses Go's standard net.ParseIP function to validate IP format and attempts to connect to port 22.
+// This function assumes IPv4 addresses are being validated.
+// Validation rules:
+// - Must be a valid IPv4 address in dotted decimal notation (e.g., "192.168.1.1")
+// - Must be reachable on port 22 (SSH)
+//
+// Returns:
+// - nil if value is a valid and reachable IP address
+// - errInvalidIPFormat if value is not a valid IP address
+// - dynamic error if IP is valid but host is unreachable (includes obfuscated IP)
+func validateIP(value string) error {
+	ip := net.ParseIP(value)
+	if ip != nil {
+		// IP format is valid, now check reachability
+		if err := ec2Helpers.PingIPAddress(value); err != nil {
+			obfuscatedIP := obfuscateIP(value)
+			return fmt.Errorf("IP address is valid but host %s is unreachable", obfuscatedIP)
+		}
```

Comment:
If we perform this here, MPC will not be able to start if at least one machine is unreachable. I think it's valuable to validate the format, but I'd suggest not to check VM's reachability while parsing the config.